src/core/search/indices.cc

```diff
@@ -123,6 +123,15 @@ const typename BaseStringIndex<C>::Container* BaseStringIndex<C>::Matching(strin
   return (it != entries_.end()) ? &it->second : nullptr;
 }
 
+template <typename C>
+void BaseStringIndex<C>::MatchingPrefix(std::string_view prefix,
+                                        absl::FunctionRef<void(const Container*)> cb) const {
+  for (auto it = entries_.lower_bound(prefix);
+       it != entries_.end() && (*it).first.rfind(prefix, 0) == 0; ++it) {
+    cb(&(*it).second);
+  }
+}
```

Comment:
I don't like the costly rfind=starts_with check, the better option is to get the next lexigocraphically larger string and iterate up it it 🤔 i.e. `abc*` -> right iterator points to lower_bound(`abd`). There might be issues with unicode (how does prefix search generally work with them 🤔), but we don't need it to be valid codepoints just to perform comparisons